azure/examples/containers/index.ts

```diff
@@ -0,0 +1,243 @@
+
+    constructor(name: string, memory: number = 128) {
+        let redis = new cloud.Service(name, {
+            containers: {
+                redis: {
+                    image: "redis:alpine",
+                    memory: memory,
+                    ports: [{ port: 6379, external: true }],
+                    command: ["redis-server", "--requirepass", redisPassword],
+                },
+            },
+        });
+        this.get = (key: string) => {
+            return redis.getEndpoint("redis", 6379).then(endpoint => {
+                console.log(`Endpoint: ${JSON.stringify(endpoint)}`);
+                let client = require("redis").createClient(
+                    endpoint.port,
+                    endpoint.hostname,
+                    { password: redisPassword },
+                );
+                console.log(client);
+                return new Promise<string>((resolve, reject) => {
+                    client.get(key, (err: any, v: any) => {
+                        if (err) {
+                            reject(err);
+                        } else {
+                            resolve(v);
+                        }
+                    });
+                });
+            });
+        };
+        this.set = (key: string, value: string) => {
+            return redis.getEndpoint("redis", 6379).then(endpoint => {
+                console.log(`Endpoint: ${JSON.stringify(endpoint)}`);
+                let client = require("redis").createClient(
+                    endpoint.port,
+                    endpoint.hostname,
+                    { password: redisPassword },
+                );
+                console.log(client);
+                return new Promise<void>((resolve, reject) => {
+                    client.set(key, value, (err: any, v: any) => {
+                        if (err) {
+                            reject(err);
+                        } else {
+                            resolve();
+                        }
+                    });
+                });
+            });
+        };
+    }
+}
+
+let cache = new Cache("examples-mycache");
+
+let helloTask = new cloud.Task("examples-hello-world", {
+    image: "hello-world",
+    memory: 20,
+});
+
+// build an anonymous image:
+let builtService = new cloud.Service("examples-nginx2", {
+    containers: {
+        nginx: {
+            build: "./app",
+            memory: 128,
+            ports: [{ port: 80, external: true }],
+        },
+    },
+    replicas: 2,
+    waitForSteadyState: false,
+});
+
+// expose some APIs meant for testing purposes.
+// let api = new cloud.API("examples-containers");
+// api.get("/test", async (req, res) => {
+//     try {
+//         res.json({
+//             nginx: await nginx.getEndpoint(),
+//             nginx2: await builtService.getEndpoint(),
+//         });
+//     } catch (err) {
+//         console.error(errorJSON(err));
+//         res.status(500).json(errorJSON(err));
+//     }
+// });
```

Comment:
commented out because we don't have API working yet.  I will bring online when i do API next.